### PR TITLE
feat: Add TF2E platform specific player counts and team players left

### PR DESCRIPTION
**Pull request by lwaddicor**
This adds support for the v7 platform specific player counts as well as the  v6  field "Teams Left With Players Num".

Players are stored as part of a map of platform name  to playercount 

Implementation spec here: https://multiplay.atlassian.net/browse/ME-1067

### CODE REVIEW

File: lib/svrquery/protocol/titanfall/query_test.go [hunk 1]
```diff
@@ -41,31 +41,40 @@ var (
...
 		},
 		MatchState: MatchState{
-			Phase:            2,
-			MaxRounds:        1,
-			RoundsWonIMC:     0,
-			RoundsWonMilitia: 0,
-			TimeLimit:        1800,
-			TimePassed:       0,
-			MaxScore:         50,
+			MatchStateV2: MatchStateV2{
```
Comment (stevenh) [inline lib/svrquery/protocol/titanfall/query_test.go:50]: Not sure why this change as they look identical, could you clarify?

File: lib/svrquery/protocol/titanfall/consts.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ const (
 	ServerInfoResponse = byte(78)
 
 	// ServerInfoVersion is the version of a info packets.
-	ServerInfoVersion = byte(3)
+	ServerInfoVersion = byte(7)
```
Comment (stevenh) [inline lib/svrquery/protocol/titanfall/consts.go:11]: Is this backwards compatible and does it need to be?

File: lib/svrquery/protocol/titanfall/query_test.go (hunk 1 above)
Comment (lwaddicor) [reply lib/svrquery/protocol/titanfall/query_test.go:50]: MatchState supports an additional TeamsLeftWithPlayersNum field (which is set below for the v7 test) to enable this to still process older requests (e.g. the keyed test) I have updated the MatchState object to include it, and embedded the older MatchStateV2 inside.

File: lib/svrquery/protocol/titanfall/query.go [hunk 3]
```diff
@@ -76,10 +76,18 @@ func (q *queryer) Query() (protocol.Responser, error) {
...
+		if i.Version > 5 {
+			// MatchState and Teams.
+			if err = r.Read(&i.MatchState); err != nil {
+				return nil, err
+			}
+		} else {
+			if err = r.Read(&i.MatchState.MatchStateV2); err != nil {
+				return nil, err
+			}
+		}
```
[the remark this answers is not in the document]
Comment (lwaddicor) [reply lib/svrquery/protocol/titanfall/query.go:88]: Done

File: lib/svrquery/protocol/titanfall/consts.go (hunk 2 above)
Comment (lwaddicor) [reply lib/svrquery/protocol/titanfall/consts.go:11]: It wont be backwards compatible becouse this is the value used in the request. The server binaries will need to support v7 for this to work when merged.

Comment (stevenh) [reply lib/svrquery/protocol/titanfall/consts.go:11]: Should we add a new protocol for the new version that way it’s config instead of code

Comment (lwaddicor) [reply lib/svrquery/protocol/titanfall/consts.go:11]: Yeah this is a good idea, i'll do this

**Comment by lwaddicor** [comment]
Retested on r5 machine, current "tf2e" works as expected, "tf2e-v7" returns no response as expected also right now.

```
lewiswa@host-103-102-161-238:~> ./cli -proto tf2e -addr "103.102.161.238:37016"
{
        "Prefix": -1,
        "Command": 78,
        "Version": 3,
        "Retail": 1,
        "InstanceType": 2,
        "ClientCRC": 4294967295,
        "NetProtocol": 542,
        "RandomServerID": 0,
        "BuildName": "R5pc_r5launch_J2583_CL552044_2020_07_08_18_06",
        "Datacenter": "japan west",
        "GameMode": "survival",
        "Port": 37015,
        "Platform": "PS4",
        "PlaylistVersion": "",
        "PlaylistNum": 516,
        "PlaylistName": "des_mu1",
        "NumClients": 14,
        "MaxClients": 64,
        "Map": "mp_rr_desertlands_mu1",
        "PlatformPlayers": null,
        "AverageFrameTime": 0,
        "MaxFrameTime": 0,
        "AverageUserCommandTime": 0,
        "MaxUserCommandTime": 0,
        "Phase": 3,
        "MaxRounds": 1,
        "RoundsWonIMC": 0,
        "RoundsWonMilitia": 0,
        "TimeLimit": 1800,
        "TimePassed": 944,
        "MaxScore": 50,
        "TeamsLeftWithPlayersNum": 0,
        "Teams": null,
        "Clients": null
}
lewiswa@host-103-102-161-238:~> ./cli -proto tf2e-v7 -addr "103.102.161.238:37016"
read udp 103.102.161.238:48610->103.102.161.238:37016: i/o timeout
lewiswa@host-103-102-161-238:~> ./cli -proto tf2e-v9 -addr "103.102.161.238:37016"
unknown protocol "tf2e-v9"
```
